cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -1,33 +1,48 @@
-from random import sample
-
 import pytest
+from cfme.middleware import get_random_list
 from cfme.middleware.deployment import MiddlewareDeployment
+from cfme.middleware.server import MiddlewareServer
 from utils import testgen
 from utils.version import current_version
 
 pytestmark = [
-    pytest.mark.usefixtures('single_middleware_provider'),
+    pytest.mark.usefixtures('setup_provider'),
```

Comment:
@hhovsepy 
Yes, of course, that is the purpose of testgen ;)

*Can we rely on UI?*
I guess that not 100%, but that's why we test it in the first place :)

So you use testgen to parametrize the tests on providers and then base the tests off the provider, so you don't care about the other influences. That is why eg. our VM objects first ask self.provider to load the details and then it clicks through the relationship to see only the VMs of that provider so you are 100% sure you only see the vms of that provider when you are looking for that particular VM.